### ISSUE TITLE
Feature Request: Libre.fm support

### ISSUE THREAD

**Issue by Phate6660** [InputUsername/rescrobbled#41]
This looks like a great project, and I would love to use it.
I use libre.fm however and it doesn't seem to be supported.

I'd be more than happy to try my hand at implementing it myself in a PR as well.
I'm just not sure how quickly I'll be able to get to it. 
I'm pretty busy with IRL things as well as various projects.

As a side note: Once the feature gets implemented and I start using it 
(I'll most likely end up using it daily), I will also be glad to create Gentoo ebuilds for this.

**Comment by InputUsername** [comment]
Thank you for your interest in the project!

I'd be glad to help you out to create a PR for Libre.fm support.
Rescrobbled uses [rustfm-scrobble](https://github.com/dmfutcher/rustfm-scrobble) for Last.fm scrobbling, and it seems there is an [open issue](https://github.com/dmfutcher/rustfm-scrobble/issues/24) to add Libre.fm support to it.

The issue has been open for a long time, perhaps you could help move it forward? Unfortunately I don't have time to do so myself at the moment.

>As a side note: Once the feature gets implemented and I start using it
>(I'll most likely end up using it daily), I will also be glad to create Gentoo ebuilds for this.
>
>Edit: I'd also be happy to make an OpenRC service too.

Both would be very welcome!

**Comment by InputUsername** [comment]
Not sure if you've had a look at rescrobbled code itself yet, but wanted to give a heads-up that I've added some changes on the [`development`](https://github.com/InputUsername/rescrobbled/tree/development) branch that should make it a bit easier to add support for Libre.fm.

**Comment by shymega** [comment]
Hey, I saw you mentioned my issue on `rustfm-scrobble`. I had some data loss when I last worked on it - it even affected my college work, lost a fair few assignments! - but I still mean to add Libre.fm support. It's also necessary for my modular scrobbler (backed by SQLite, very similar to this one), so that was my reason for implementing Libre.fm support.

I also need to write a crate for Listenbrainz, which you may be interested in.

**Comment by shymega** [comment]
See: https://github.com/dmfutcher/rustfm-scrobble/issues/24#issuecomment-885615632

**Comment by InputUsername** [comment]
@shymega I've got an existing [`listenbrainz`](https://github.com/InputUsername/listenbrainz-rs) crate, but I agree that turning rustfm-scrobble into a unified scrobbler sounds appealing as well.

**Comment by shymega** [comment]
@InputUsername Yeah, I'll wait for the author of `rustfm-scrobble` to get back to me. I think that crate will need a bit of refactoring though. Depends on what the author wants to do... or we could make a fork?

The problem really is that its very much designed for Last.fm, and whilst Libre.fm would generally fit in quite well, Listenbrainz is a very different beast!

Thanks for the quick reply... :+1:
